### ISSUE TITLE
PMP do_intersect compiler errors when using Epeck kernel

### ISSUE THREAD

**Issue by DABH** [CGAL/cgal#3403]
## Issue Details

I'm trying to make a call like
`bool overlap = PMP::do_intersect(a, b);`
where `a` and `b` both have type `CGAL::Surface_mesh<CGAL::Exact_predicates_exact_constructions_kernel::Point_3>`.  Looks like this should work according to the docs (https://doc.cgal.org/latest/Polygon_mesh_processing/group__PMP__predicates__grp.html#ga2ee87a0a4368a21654ee2817b93c85e4).

However, I get these compiler errors:
```
/usr/local/include/CGAL/Boolean_set_operations_2.h: In instantiation of ‘struct CGAL::map_iterator_to_traits<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epeck> > >’:
/usr/local/include/CGAL/Boolean_set_operations_2.h:1048:60:   required from ‘bool CGAL::do_intersect(InputIterator1, InputIterator1, InputIterator2, InputIterator2, unsigned int) [with InputIterator1 = CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epeck> >; InputIterator2 = CGAL::cgal_bgl_named_params<bool, CGAL::internal_np::all_default_t>]’
/usr/local/include/CGAL/Polygon_mesh_processing/intersection.h:1254:22:   required from ‘bool CGAL::Polygon_mesh_processing::do_intersect(const TriangleMesh&, const TriangleMesh&, const typename boost::disable_if<typename boost::has_range_const_iterator<Polyline>::type>::type*) [with TriangleMesh = CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epeck> >; typename boost::disable_if<typename boost::has_range_const_iterator<Polyline>::type>::type = void]’
/usr/local/include/CGAL/Boolean_set_operations_2.h:846:68: error: no type named ‘value_type’ in ‘struct std::iterator_traits<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epeck> > >’
   typedef typename std::iterator_traits<InputIterator>::value_type InputPolygon;
                                                                    ^~~~~~~~~~~~
/usr/local/include/CGAL/Boolean_set_operations_2.h:847:64: error: no type named ‘value_type’ in ‘struct std::iterator_traits<CGAL::Surface_mesh<CGAL::Point_3<CGAL::Epeck> > >’
   typedef typename Gps_default_traits<InputPolygon>::Traits    Traits;
                                                                ^~~~~~
```
which is weird that a seemingly 2D file is being referred to.

In any case, I'd appreciate any help in pointing out what I did wrong, or figuring out what the bug is (maybe Epeck kernel isn't supported yet for this function?).  Thanks a lot for your help!

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Linux x64
* Compiler: g++ 6.3.0
* Release or debug mode: Release
* Specific flags used (if any):
* CGAL version: master
* Boost version: 1.62
* Other libraries versions if used (Eigen, TBB, etc.):

**Comment by MaelRL** [comment]
`do_intersect` works with `EPECK`, it is tested [here](https://github.com/CGAL/cgal/blob/master/Polygon_mesh_processing/test/Polygon_mesh_processing/pmp_do_intersect_test.cpp#L404) for example.

There's some issue with the look-up, and it's ending up in `Boolean_set_operations_2.h` because this package also offers an (unrelated) `do_intersect` function. Could you please provide your exact function call?

**Comment by MaelRL** [comment]
There's indeed a problem, thanks. It will be fixed in https://github.com/CGAL/cgal/pull/3407.

**Comment by lrineau** [comment]
Fixed by #3407.
